fast_tsne.m

```diff
@@ -47,6 +47,17 @@
 % IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY 
 % OF SUCH DAMAGE.
 
+    % rotate the input so that it makes sense
+    if size(X,1) < size(X,2)
+        X = X';
+    end

```

Comment:
I don't think this is a good idea. Some people may have situations in which D > N, and then run into weird problems.